src/Tizen.Sensor/Tizen.Sensor/SensorProvider.cs

```diff
@@ -0,0 +1,467 @@
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class SensorProvider : IDisposable
+    {
+        internal IntPtr _handle;
+        private static Interop.SensorManager.SensorAddedCb _sensorAddedCallBack = null;
+        private static Interop.SensorManager.SensorRemovedCb _sensorRemovedCallBack = null;
+        private static Interop.SensorProvider.SensorProviderStartCb _sensorProviderStartCallBack = null;
+        private static Interop.SensorProvider.SensorProviderStopCb _sensorProviderStopCallBack = null;
+        private static Interop.SensorProvider.SensorProviderIntervalChangedCb _sensorProviderIntervalChangedCallBack = null;
+        /// <summary>
+        /// ProviderIntervalChanged
+        /// </summary>
+        private static event EventHandler<ProviderIntervalChangedEventArgs> ProviderIntervalChanged;
+
+        /// <summary>
+        /// SensorProvider constructor.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="handle"> The account handle.</param>
+        internal SensorProvider(IntPtr handle) {
+            Handle = handle;
+        }
+
+        /// <summary>
+        /// SensorProvider deconstructor.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        ~SensorProvider()
+        {
+            Dispose(false);
+        }
+
+        internal IntPtr Handle
+        {
+            get
+            {
+                return _handle;
+            }
+
+            set
+            {
+                _handle = value;
+            }
+        }
+
+        internal bool CheckProviderHandle()
+        {
+            bool result = false;
+            if (Handle != IntPtr.Zero)
+            {
+                result = true;
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Sensor Provider handle is null");
+                throw new ArgumentException("Invalid Parameter: Sensor provider handle is null");
+            }
+            return result;
+        }
+
+        /// <summary>
+        ///  This function creates a sensor provider handle with a given URI.
+        /// </summary>
+        /// <param name="uri">
+        /// The URI of sensor to be created.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public SensorProvider CreateProvider(string uri)
+        {
+            IntPtr handle;
+            int error = Interop.SensorProvider.CreateProvider(uri, out handle);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in creating sensor provider");
+                throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.CreateProvider Failed");
+            }
+            return new SensorProvider(handle);
+        }
+
+
+        /// <summary>
+        /// Registers the sensor provider.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void AddProvider(SensorProvider handle)
```

Comment:
```suggestion
        public void Start()
```

Reply:
Sir, Do you mean the handle created in create provider will be used in Start(). and no need to add handle as parameter.

Reply:
Yes, no need to pass handle because `_handle` should be used